tiledb/sm/group/group.cc

```diff
@@ -784,11 +796,24 @@ std::string Group::dump(
         member_uri = group_uri_.join_path(it->uri().to_string());
       }
 
+      // Check if we've already visited this group to avoid cycles
+      std::string member_uri_str = member_uri.to_string();
+      if (visited.find(member_uri_str) != visited.end()) {
+        ss << " (cycle detected)" << std::endl;
```

Comment:
`(see above)` sounds better, but I'd also be fine with displaying no additional text.